packages/compiler/react/visitor.ts

```diff
@@ -180,27 +186,34 @@ export const visitor = (options: Options = {}, isReact = true) => {
 
     callSitePath.scope.crawl();
 
-    const componentDeclarationPath = callSitePath.scope.getBinding(
+    let componentDeclarationPath = callSitePath.scope.getBinding(
       isComponentAnonymous
         ? // We know that the component is a identifier, so we can safely cast it.
           (callSite.arguments[0] as t.Identifier).name
         : RawComponent.name,
     )!.path;
 
+    if (componentDeclarationPath.isImportSpecifier()) {
+      const correctComponentDeclarationPath = getSpecifierSource(
+        componentDeclarationPath,
+        options.id!,
+      );
+      if (correctComponentDeclarationPath) {
+        componentDeclarationPath.remove();
+        globalPath.insertBefore(correctComponentDeclarationPath.node);
```

Comment:
What happens when the imported component depends on external scope variables in the original file? For instance:
```tsx
// in original.tsx
const externalVariable = 'Hello';
export const Component = () => <div>{externalVariable}</div>;

// in different-file.tsx
import { Component } from './original';
const ComponentBlock = block(Component);
```